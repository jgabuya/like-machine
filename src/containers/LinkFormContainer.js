import React, { Component } from 'react';
import { connect } from 'react-redux';
import LinkForm from '../components/LinkForm';
import postLink from '../store/actions/post-link';

class LinkFormContainer extends Component {
    constructor(props) {
        super(props);

        this.state = {
            url: '' 
        }

        this.onUrlChange = this.onUrlChange.bind(this);
    }

    onUrlChange(e) {
        this.setState({ url: e.target.value })
    }    

    render() {
        return (
            <LinkForm
                onUrlChange={this.onUrlChange}
                onSubmit={(e) => {
                    e.preventDefault();
                    this.props.onSubmit(this.state);
                }}
            />
        );
    }
}

const mapStateToProps = state => state;

const mapDispatchToProps = {
	onSubmit: postLink
};

export default connect(mapStateToProps, mapDispatchToProps)(LinkFormContainer);