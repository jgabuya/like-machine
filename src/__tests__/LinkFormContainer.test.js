import React from "react";
import LinkFormContainer from "../containers/LinkFormContainer";
import { shallow } from "enzyme";
import thunk from "redux-thunk";
import configureStore from "redux-mock-store";
import renderer from "react-test-renderer";

describe("LinkFormContainer", () => {
    let component;
    let wrapper;
    let store;

    const initialState = {
        onUrlInputChange: jest.fn(),
        formData: { url: "" },
        user: { name: "" }
    };

    beforeEach(() => {
        store = configureStore([thunk])(initialState);
        component = <LinkFormContainer store={store} />;
        wrapper = shallow(component);
    });

    it("renders without crashing", () => {
        expect(wrapper.length).toBe(1);
    });

    it("matches snapshot", () => {
        const tree = renderer.create(component).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
