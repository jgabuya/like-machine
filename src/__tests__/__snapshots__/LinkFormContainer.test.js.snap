// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkFormContainer matches snapshot 1`] = `
<div>
  <p
    className="text-muted"
  >
    Welcome, 
    
  </p>
  <div>
    <form
      className=""
      onSubmit={[Function]}
    >
      <h4>
        New Link
      </h4>
      <div
        className="position-relative form-group"
      >
        <label
          className=""
          htmlFor="input-title"
        >
          URL
        </label>
        <input
          className="form-control"
          id="input-title"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="position-relative form-group"
      >
        <button
          className="btn btn-primary"
          onClick={[Function]}
          type={undefined}
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</div>
`;
